docker-compose.yml

```diff
@@ -6,14 +6,13 @@ services:
     restart: unless-stopped
     ports:
       - ${CORE_P2P_PORT:?err}:${CORE_P2P_PORT:?err} # P2P
+      - 127.0.0.1:${CORE_RPC_PORT:?err}:${CORE_RPC_PORT:?err} #RPC
     volumes:
-      - ./configs/${NETWORK:?err}/core/dashd.conf:/dash/.dashcore/dash.conf
+      - ~/.mn/${CONFIG_NAME:?err}/dashd.conf:/dash/.dashcore/dash.conf
       - core_data:/dash
     command:
       - dashd
-      - -port=${CORE_P2P_PORT:?err}
       - -masternodeblsprivkey=${CORE_MASTERNODE_OPERATOR_PRIVATE_KEY}
```

Comment:
Should we set it in the dashd config too?

Reply:
Setting this in dashd config means functions that require the wallet like `registerMasternodeTask` and `generateToAddressTask` don't work because the masternode is still enabled in the conf file:

```
Error: You can not start a masternode with wallet enabled.
```

I'm trying to temporarily do `config.set('core.masternode.enable', false);` and set it back to its original state later, but I don't understand when the service templates are actually rendered. Is it possible to use a debugger to see the actual sequence in which the functions run? Is this what the call stack is? Does run() or or runWithDependencies() go first, or are they the same thing?

Reply:
`run` goes first and then calls `runWithDependencies`. It seems it's better to move it back to docker-compose.

Reply:
OK, finally resolved this, see comments below on related topics.